src/pybind/mgr/volumes/fs/operations/trash.py

```diff
@@ -122,6 +122,50 @@ def delink(self, bname):
         except cephfs.Error as e:
             raise VolumeException(-e.args[0], e.args[1])
 
+    def _get_stats(self, path, num_of_files, num_of_subvols, iter_level):
+        with self.fs.opendir(path) as dir_handle:
+            d = self.fs.readdir(dir_handle)
+            while d:
+                if d.d_name not in (b'.', b'..'):
+                    if d.is_dir():
+                        d_full = os.path.join(path, d.d_name)
+                        iter_level += 1
+                        num_of_files, num_of_subvols = self._get_stats(d_full,
+                            num_of_files, num_of_subvols, iter_level)
+                        iter_level -= 1
+
+                    num_of_files += 1
+                    if iter_level == 0:
+                        num_of_subvols += 1
+
+                try:
+                    d = self.fs.readdir(dir_handle)
+                # this try-except is inside the loop so that looping can
+                # "continue" in ObjectNotFound
+                except cephfs.ObjectNotFound as e:
+                    log.debug(f'Exception "{e}" occurred, perhaps the file '
+                              'purged by purge threads that are running '
+                              'simultaneously')
+                    continue
+
+        return num_of_files, num_of_subvols
+
+    def get_stats(self):
+        try:
+            num_of_files, num_of_subvols = self._get_stats(self.path, 0, 0, 0)
+        # this try-except ensure separate handling when trash dir goes
+        # missing
+        except cephfs.ObjectNotFound as e:
```

Comment:
As commented in other commits, catch cephfs.Error and then compare ENOENT

Reply:
Made the change.